pkg/models/terminal/terminal.go

```diff
@@ -140,15 +149,133 @@ func (t TerminalSession) Close(status uint32, reason string) {
 
 type Interface interface {
 	HandleSession(shell, namespace, podName, containerName string, conn *websocket.Conn)
+	HandleShellAccessToNode(nodename string, conn *websocket.Conn)
 }
 
 type terminaler struct {
-	client kubernetes.Interface
-	config *rest.Config
+	client  kubernetes.Interface
+	config  *rest.Config
+	options *Options
+}
+
+type NodeTerminaler struct {
+	Nodename      string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Shell         string
+	Privileged    bool
+	Config        *Options
+	client        kubernetes.Interface
 }
 
-func NewTerminaler(client kubernetes.Interface, config *rest.Config) Interface {
-	return &terminaler{client: client, config: config}
+func NewTerminaler(client kubernetes.Interface, config *rest.Config, options *Options) Interface {
+	return &terminaler{client: client, config: config, options: options}
+}
+
+func NewNodeTerminaler(nodename string, options *Options, client kubernetes.Interface) (*NodeTerminaler, error) {
+
+	n := &NodeTerminaler{
+		Namespace:     "kubesphere-controls-system",
+		ContainerName: "nsenter",
+		Nodename:      nodename,
+		PodName:       nodename + "-shell-access",
+		Shell:         "sh",
+		Privileged:    true,
+		Config:        options,
+		client:        client,
+	}
+
+	node, err := n.client.CoreV1().Nodes().Get(context.Background(), n.Nodename, metav1.GetOptions{})
+
+	if err != nil {
+		return n, fmt.Errorf("node cannot exist. nodename:%s, err: %v", n.Nodename, err)
+	}
+
+	flag := false
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+			flag = true
+			break
+		}
+	}
+	if !flag {
+		return n, fmt.Errorf("node status error. node: %s", n.Nodename)
+	}
+
+	idx := int64(0)
+	NodeSessionCounter.LoadOrStore(nodename, &idx)
```

Comment:
Correct me if I am wrong. `NodeSessionCounter` is used to store user sessions, if there is more than one session attached to the exec pod,  we don't clean the exec pod. Since `NodeSessionCounter` is stored in memory, isn't data lost between ks-apiserver restarts?

Reply:
Maybe need to check the pod on this node is not existed before setting it to zero.

Reply:
Does this logic can be removed ?  And combines the initial logic in here:
```
idx, _ := NodeSessionCounter.Load(n.Nodename)
atomic.AddInt64(idx.(*int64), 1)
close(succ)
```
If we load successfully, let's add one to the value.  Or we directly set it to 1. This will save once loadorstore in memory.
What do you think @lynxcat @zryfish